analyses/mutational-signatures/08-explore_hypermutators.Rmd

```diff
@@ -143,6 +150,78 @@ A couple interesting observations in the table above:
 
 + The HGG hypermutator sample with lowest TMB (TMB=15.48) is primarily exposed to signature 1, which is a normal signature.
 
+### Make heatmap for mutational signatures for patients with hypermutant tumors
+```{r}
+# Create signature matrix for only patients withh hypermutant samples
+# Gather all BS_IDs (soi = "samples of interest"), whether hypermutant or not, for pts with hypermutant samples
+
+soi_df <- meta %>%
+  filter(Kids_First_Biospecimen_ID %in% hypermutator_df$Kids_First_Biospecimen_ID) %>%
+  select(Kids_First_Participant_ID) %>%
+  left_join(meta) %>%
+  #keep only WGS samples; remove normals and cell lines
+  filter(experimental_strategy == "WGS" & !is.na(pathology_diagnosis) & composition != "Derived Cell Line") %>%
+  select(Kids_First_Participant_ID, Kids_First_Biospecimen_ID, tumor_descriptor) %>%
+  unique() %>%
+  arrange(Kids_First_Participant_ID)
+         
+sigs <- exposures %>%
+  filter(signature != "Other") %>%
+  tidyr::spread(signature, exposure) %>%
+  select(-broad_histology, -cancer_group_display_n, -n, -cancer_group_display, -cancer_group_hex, -tumor_descriptor) %>%
+  filter(Kids_First_Biospecimen_ID %in% soi_df$Kids_First_Biospecimen_ID) %>%
+  as.data.frame()
+rownames(sigs) <- sigs$Kids_First_Biospecimen_ID
+
+sigs$Kids_First_Biospecimen_ID <- NULL
+sigs <- as.matrix(sigs)
+sapply(sigs, class)
+
+# Create annotation dataframe
+anno <- tmb_coding %>%
+  mutate(Kids_First_Biospecimen_ID = Tumor_Sample_Barcode) %>%
+  select(Kids_First_Biospecimen_ID, tmb) %>%
+  # add a column for mutator
+  mutate(
+    Mutator = case_when(
+      tmb < 10 ~ "Normal",
+      tmb >= 10 & tmb < 100 ~ "Hypermutant",
+      tmb >= 100 ~ "Ultra-hypermutant")
+  ) %>%
+  right_join(soi_df, by = "Kids_First_Biospecimen_ID") %>%
+  unique() %>%
+  # mutate names for annotation
+  mutate(Patient = Kids_First_Participant_ID,
+         `Phase of Therapy` = tumor_descriptor) %>%
+  select(Patient, Kids_First_Biospecimen_ID, `Phase of Therapy`, Mutator) %>%
+  as.data.frame()
+
+# set rownames to match matrix, remove BS_ID
+rownames(anno) <- anno$Kids_First_Biospecimen_ID
+anno$Kids_First_Biospecimen_ID <- NULL
+
+# Specify colors
+ann_colors = list(
+  # mutator color from [this PR](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1280/files#diff-bc8134008706f38e7795b59696928058d123e44b4172df68105a26ad8aa0a398R298-R301) 
+  Mutator = c(Normal = "grey40",
+              Hypermutant = "orange",
+              `Ultra-hypermutant` = "red"),
+  # tumor descriptor colors from [palette](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/figures/palettes/tumor_descriptor_palette.tsv) 
+  `Phase of Therapy` = c(`Initial CNS Tumor` = "#709AE1", 
```

Comment:
Rather than setting these colors manually, we should probably read in the file `figures/palettes/tumor_descriptor_palette.tsv` and extract colors from that data frame.

Reply:
updated!